src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,128 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { getFilmByTitle } from '../../API/api';
+import { MovieCard } from '../MovieCard';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+type Props = {
+  selectedMovie: Dispatch<SetStateAction<Movie[]>>,
+  movies: Movie[],
+};
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie: React.FC<Props> = ({ selectedMovie, movies }) => {
+  const [title, setTitle] = useState('');
+  const [preview, setPreview] = useState<Movie | null>(null);
+  const [error, setError] = useState(false);
+  const [duplicate, setDuplicate] = useState(false);
+
+  const findMovie = () => {
+    getFilmByTitle(title)
+      .then(film => {
+        if (film.Error) {
+          setError(true);
+        } else {
+          return setPreview(film);
+        }
+
+        return null;
+      });
+  };
+
+  const addmovie = () => {
+    if (preview
+      && movies.every(movie => movie.imdbID !== preview.imdbID)) {
```

Comment:
I would use `.findindex` or `.indexOf` instead of `.every`. Couse `.every` will do iteration to the end of the array but the object which you try to find in the midle /start.